py/requirements.txt

```diff
@@ -2,7 +2,8 @@ numpy
 packaging
 pybind11==2.6.2
 --extra-index-url https://download.pytorch.org/whl/nightly/cu124
-torch>=2.5.0.dev,<2.6.0
+# it is trying to download all the torch versions from 0808 back to 0617 to find the matching torchvision 0.19.0 which cause disk space to be full
+torch>=2.5.0.dev20240617,<2.5.0.dev20240618
```

Comment:
Can we get rid of this file and just use pyproject.toml?

Reply:
I prefer change this in a separate PR, because it might need to make changes on the windows/linux build/test/release scripts.
So that this PR can be focused on the docker change only.

Reply:
limit torch version to torch>=2.5.0.dev20240617,<2.5.0.dev20240618 is not ideal, it should be temporary.
we should try to find a solution of bump torchvision version to 0.20.0
currently if bump torchvision to 0.20.0 we have failed testcases: 
Some arguments (weight, bias, running_mean, running_var) to batch_norm are passed with different types between torchvision 0.19 and 0.20 in Test torch compile backend CI.
https://github.com/pytorch/TensorRT/actions/runs/10224911783/job/28582163503?pr=3001